<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>_SummarizedExperiment_ for Coordinating Experimental Assays, Samples, and Regions of Interest • SummarizedExperimentGHA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="_SummarizedExperiment_ for Coordinating Experimental Assays, Samples, and Regions of Interest">
<meta property="og:description" content="SummarizedExperimentGHA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SummarizedExperimentGHA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.23.5003</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Extensions.html">Extending the SummarizedExperiment class</a>
    </li>
    <li>
      <a href="../articles/SummarizedExperiment.html">_SummarizedExperiment_ for Coordinating Experimental Assays, Samples, and Regions of Interest</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/vjcitn/SummarizedExperimentGHA">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="SummarizedExperiment_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>
<em>SummarizedExperiment</em> for Coordinating Experimental Assays, Samples, and Regions of Interest</h1>
                        <h4 class="author">Martin Morgan, Valerie Obenchain, Jim Hester, Hervé Pagès</h4>
            
            <h4 class="date">Revised: 3 May, 2018</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vjcitn/SummarizedExperimentGHA/blob/master/vignettes/SummarizedExperiment.Rmd"><code>vignettes/SummarizedExperiment.Rmd</code></a></small>
      <div class="hidden name"><code>SummarizedExperiment.Rmd</code></div>

    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The <code>SummarizedExperiment</code> class is used to store rectangular matrices of experimental results, which are commonly produced by sequencing and microarray experiments. Note that <code>SummarizedExperiment</code> can simultaneously manage several experimental results or <code>assays</code> as long as they be of the same dimensions.</p>
<p>Each object stores observations of one or more samples, along with additional meta-data describing both the observations (features) and samples (phenotypes).</p>
<p>A key aspect of the <code>SummarizedExperiment</code> class is the coordination of the meta-data and assays when subsetting. For example, if you want to exclude a given sample you can do for both the meta-data and assay in one operation, which ensures the meta-data and observed data will remain in sync. Improperly accounting for meta and observational data has resulted in a number of incorrect results and retractions so this is a very desirable property.</p>
<p><code>SummarizedExperiment</code> is in many ways similar to the historical <code>ExpressionSet</code>, the main distinction being that <code>SummarizedExperiment</code> is more flexible in it’s row information, allowing both <code>GRanges</code> based as well as those described by arbitrary <code>DataFrame</code>s. This makes it ideally suited to a variety of experiments, particularly sequencing based experiments such as RNA-Seq and ChIp-Seq.</p>
</div>
<div id="anatomy-of-a-summarizedexperiment" class="section level1">
<h1 class="hasAnchor">
<a href="#anatomy-of-a-summarizedexperiment" class="anchor"></a>Anatomy of a <code>SummarizedExperiment</code>
</h1>
<p>The <em>SummarizedExperiment</em> package contains two classes: <code>SummarizedExperiment</code> and <code>RangedSummarizedExperiment</code>.</p>
<p><code>SummarizedExperiment</code> is a matrix-like container where rows represent features of interest (e.g. genes, transcripts, exons, etc.) and columns represent samples. The objects contain one or more assays, each represented by a matrix-like object of numeric or other mode. The rows of a <code>SummarizedExperiment</code> object represent features of interest. Information about these features is stored in a <code>DataFrame</code> object, accessible using the function <code><a href="../reference/SummarizedExperiment-class.html">rowData()</a></code>. Each row of the <code>DataFrame</code> provides information on the feature in the corresponding row of the <code>SummarizedExperiment</code> object. Columns of the DataFrame represent different attributes of the features of interest, e.g., gene or transcript IDs, etc.</p>
<p><code>RangedSummarizedExperiment</code> is the child of the <code>SummarizedExperiment</code> class which means that all the methods on <code>SummarizedExperiment</code> also work on a <code>RangedSummarizedExperiment</code>.</p>
<p>The fundamental difference between the two classes is that the rows of a <code>RangedSummarizedExperiment</code> object represent genomic ranges of interest instead of a <code>DataFrame</code> of features. The <code>RangedSummarizedExperiment</code> ranges are described by a <code>GRanges</code> or a <code>GRangesList</code> object, accessible using the <code><a href="../reference/RangedSummarizedExperiment-class.html">rowRanges()</a></code> function.</p>
<p>The following graphic displays the class geometry and highlights the vertical (column) and horizontal (row) relationships.</p>
<div class="figure">
<img src="SE.svg" alt=""><p class="caption">Summarized Experiment</p>
</div>
<div id="assays" class="section level2">
<h2 class="hasAnchor">
<a href="#assays" class="anchor"></a>Assays</h2>
<p>The <code>airway</code> package contains an example dataset from an RNA-Seq experiment of read counts per gene for airway smooth muscles. These data are stored in a <code>RangedSummarizedExperiment</code> object which contains 8 different experimental and assays 64,102 gene transcripts.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://vjcitn.github.io/SummarizedExperimentGHA">SummarizedExperimentGHA</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">airway</span>, package<span class="op">=</span><span class="st">"airway"</span><span class="op">)</span>
<span class="va">se</span> <span class="op">&lt;-</span> <span class="va">airway</span>
<span class="va">se</span></code></pre></div>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Registered S3 methods overwritten by 'SummarizedExperiment':
##   method                                  from                   
##   .DollarNames.SummarizedExperiment       SummarizedExperimentGHA
##   .DollarNames.RangedSummarizedExperiment SummarizedExperimentGHA</code></pre>
<pre><code>## 
## Attaching package: 'SummarizedExperiment'</code></pre>
<pre><code>## The following objects are masked from 'package:SummarizedExperimentGHA':
## 
##     .DollarNames.RangedSummarizedExperiment,
##     .DollarNames.SummarizedExperiment, assay, assay&lt;-, assayNames,
##     assayNames&lt;-, assays, Assays, assays&lt;-, colData, colData&lt;-,
##     geneRangeMapper, makeSummarizedExperimentFromDataFrame,
##     makeSummarizedExperimentFromExpressionSet,
##     makeSummarizedExperimentFromLoom, naiveRangeMapper,
##     probeRangeMapper, readKallisto, readKallistoBootstrap, rowData,
##     rowData&lt;-, rowRanges&lt;-, SummarizedExperiment</code></pre>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 64102 8 
## metadata(1): ''
## assays(1): counts
## rownames(64102): ENSG00000000003 ENSG00000000005 ... LRG_98 LRG_99
## rowData names(0):
## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
<p>To retrieve the experiment data from a <code>SummarizedExperiment</code> object one can use the <code><a href="../reference/SummarizedExperiment-class.html">assays()</a></code> accessor. An object can have multiple assay datasets each of which can be accessed using the <code>$</code> operator. The <code>airway</code> dataset contains only one assay (<code>counts</code>). Here each row represents a gene transcript and each column one of the samples.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span></code></pre></div>
<table class="table">
<colgroup>
<col width="15%">
<col width="10%">
<col width="10%">
<col width="10%">
<col width="10%">
<col width="10%">
<col width="10%">
<col width="10%">
<col width="10%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">SRR1039508</th>
<th align="right">SRR1039509</th>
<th align="right">SRR1039512</th>
<th align="right">SRR1039513</th>
<th align="right">SRR1039516</th>
<th align="right">SRR1039517</th>
<th align="right">SRR1039520</th>
<th align="right">SRR1039521</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000000003</td>
<td align="right">679</td>
<td align="right">448</td>
<td align="right">873</td>
<td align="right">408</td>
<td align="right">1138</td>
<td align="right">1047</td>
<td align="right">770</td>
<td align="right">572</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000005</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000000419</td>
<td align="right">467</td>
<td align="right">515</td>
<td align="right">621</td>
<td align="right">365</td>
<td align="right">587</td>
<td align="right">799</td>
<td align="right">417</td>
<td align="right">508</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000457</td>
<td align="right">260</td>
<td align="right">211</td>
<td align="right">263</td>
<td align="right">164</td>
<td align="right">245</td>
<td align="right">331</td>
<td align="right">233</td>
<td align="right">229</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000000460</td>
<td align="right">60</td>
<td align="right">55</td>
<td align="right">40</td>
<td align="right">35</td>
<td align="right">78</td>
<td align="right">63</td>
<td align="right">76</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000938</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000000971</td>
<td align="right">3251</td>
<td align="right">3679</td>
<td align="right">6177</td>
<td align="right">4252</td>
<td align="right">6721</td>
<td align="right">11027</td>
<td align="right">5176</td>
<td align="right">7995</td>
</tr>
<tr class="even">
<td align="left">ENSG00000001036</td>
<td align="right">1433</td>
<td align="right">1062</td>
<td align="right">1733</td>
<td align="right">881</td>
<td align="right">1424</td>
<td align="right">1439</td>
<td align="right">1359</td>
<td align="right">1109</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000001084</td>
<td align="right">519</td>
<td align="right">380</td>
<td align="right">595</td>
<td align="right">493</td>
<td align="right">820</td>
<td align="right">714</td>
<td align="right">696</td>
<td align="right">704</td>
</tr>
<tr class="even">
<td align="left">ENSG00000001167</td>
<td align="right">394</td>
<td align="right">236</td>
<td align="right">464</td>
<td align="right">175</td>
<td align="right">658</td>
<td align="right">584</td>
<td align="right">360</td>
<td align="right">269</td>
</tr>
</tbody>
</table>
</div>
<div id="row-regions-of-interest-data" class="section level2">
<h2 class="hasAnchor">
<a href="#row-regions-of-interest-data" class="anchor"></a>‘Row’ (regions-of-interest) data</h2>
<p>The <code><a href="../reference/RangedSummarizedExperiment-class.html">rowRanges()</a></code> accessor is used to view the range information for a <code>RangedSummarizedExperiment</code>. (Note if this were the parent <code>SummarizedExperiment</code> class we’d use <code><a href="../reference/SummarizedExperiment-class.html">rowData()</a></code>). The data are stored in a <code>GRangesList</code> object, where each list element corresponds to one gene transcript and the ranges in each <code>GRanges</code> correspond to the exons in the transcript.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/RangedSummarizedExperiment-class.html">rowRanges</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code>## GRangesList object of length 64102:
## $ENSG00000000003
## GRanges object with 17 ranges and 2 metadata columns:
##        seqnames            ranges strand |   exon_id       exon_name
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;     &lt;character&gt;
##    [1]        X 99883667-99884983      - |    667145 ENSE00001459322
##    [2]        X 99885756-99885863      - |    667146 ENSE00000868868
##    [3]        X 99887482-99887565      - |    667147 ENSE00000401072
##    [4]        X 99887538-99887565      - |    667148 ENSE00001849132
##    [5]        X 99888402-99888536      - |    667149 ENSE00003554016
##    ...      ...               ...    ... .       ...             ...
##   [13]        X 99890555-99890743      - |    667156 ENSE00003512331
##   [14]        X 99891188-99891686      - |    667158 ENSE00001886883
##   [15]        X 99891605-99891803      - |    667159 ENSE00001855382
##   [16]        X 99891790-99892101      - |    667160 ENSE00001863395
##   [17]        X 99894942-99894988      - |    667161 ENSE00001828996
##   -------
##   seqinfo: 722 sequences (1 circular) from an unspecified genome
## 
## ...
## &lt;64101 more elements&gt;</code></pre>
</div>
<div id="column-sample-data" class="section level2">
<h2 class="hasAnchor">
<a href="#column-sample-data" class="anchor"></a>‘Column’ (sample) data</h2>
<p>Sample meta-data describing the samples can be accessed using <code><a href="../reference/SummarizedExperiment-class.html">colData()</a></code>, and is a <code>DataFrame</code> that can store any number of descriptive columns for each sample row.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 8 rows and 9 columns
##            SampleName     cell      dex    albut        Run avgLength
##              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;
## SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126
## SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126
## SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126
## SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87
## SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120
## SRR1039517 GSM1275871  N080611    trt      untrt SRR1039517       126
## SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101
## SRR1039521 GSM1275875  N061011    trt      untrt SRR1039521        98
##            Experiment    Sample    BioSample
##              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;
## SRR1039508  SRX384345 SRS508568 SAMN02422669
## SRR1039509  SRX384346 SRS508567 SAMN02422675
## SRR1039512  SRX384349 SRS508571 SAMN02422678
## SRR1039513  SRX384350 SRS508572 SAMN02422670
## SRR1039516  SRX384353 SRS508575 SAMN02422682
## SRR1039517  SRX384354 SRS508576 SAMN02422673
## SRR1039520  SRX384357 SRS508579 SAMN02422683
## SRR1039521  SRX384358 SRS508580 SAMN02422677</code></pre>
<p>This sample metadata can be accessed using the <code>$</code> accessor which makes it easy to subset the entire object by a given phenotype.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># subset for only those samples treated with dexamethasone</span>
<span class="va">se</span><span class="op">[</span>, <span class="va">se</span><span class="op">$</span><span class="va">dex</span> <span class="op">==</span> <span class="st">"trt"</span><span class="op">]</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 64102 4 
## metadata(1): ''
## assays(1): counts
## rownames(64102): ENSG00000000003 ENSG00000000005 ... LRG_98 LRG_99
## rowData names(0):
## colnames(4): SRR1039509 SRR1039513 SRR1039517 SRR1039521
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
</div>
<div id="experiment-wide-metadata" class="section level2">
<h2 class="hasAnchor">
<a href="#experiment-wide-metadata" class="anchor"></a>Experiment-wide metadata</h2>
<p>Meta-data describing the experimental methods and publication references can be accessed using <code><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html">metadata()</a></code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html">metadata</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code>## [[1]]
## Experiment data
##   Experimenter name: Himes BE 
##   Laboratory: NA 
##   Contact information:  
##   Title: RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells. 
##   URL: http://www.ncbi.nlm.nih.gov/pubmed/24926665 
##   PMIDs: 24926665 
## 
##   Abstract: A 226 word abstract is available. Use 'abstract' method.</code></pre>
<p>Note that <code><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html">metadata()</a></code> is just a simple list, so it is appropriate for <em>any</em> experiment wide metadata the user wishes to save, such as storing model formulas.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html">metadata</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">$</span><span class="va">formula</span> <span class="op">&lt;-</span> <span class="va">counts</span> <span class="op">~</span> <span class="va">dex</span> <span class="op">+</span> <span class="va">albut</span>

<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html">metadata</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code>## [[1]]
## Experiment data
##   Experimenter name: Himes BE 
##   Laboratory: NA 
##   Contact information:  
##   Title: RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells. 
##   URL: http://www.ncbi.nlm.nih.gov/pubmed/24926665 
##   PMIDs: 24926665 
## 
##   Abstract: A 226 word abstract is available. Use 'abstract' method.
## 
## $formula
## counts ~ dex + albut</code></pre>
</div>
</div>
<div id="constructing-a-summarizedexperiment" class="section level1">
<h1 class="hasAnchor">
<a href="#constructing-a-summarizedexperiment" class="anchor"></a>Constructing a <code>SummarizedExperiment</code>
</h1>
<p>Often, <code>SummarizedExperiment</code> or <code>RangedSummarizedExperiment</code> objects are returned by functions written by other packages. However it is possible to create them by hand with a call to the <code><a href="../reference/RangedSummarizedExperiment-class.html">SummarizedExperiment()</a></code> constructor.</p>
<p>Constructing a <code>RangedSummarizedExperiment</code> with a <code>GRanges</code> as the <em>rowRanges</em> argument:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nrows</span> <span class="op">&lt;-</span> <span class="fl">200</span>
<span class="va">ncols</span> <span class="op">&lt;-</span> <span class="fl">6</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="va">nrows</span> <span class="op">*</span> <span class="va">ncols</span>, <span class="fl">1</span>, <span class="fl">1e4</span><span class="op">)</span>, <span class="va">nrows</span><span class="op">)</span>
<span class="va">rowRanges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">150</span><span class="op">)</span><span class="op">)</span>,
                     <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">floor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fl">200</span>, <span class="fl">1e5</span>, <span class="fl">1e6</span><span class="op">)</span><span class="op">)</span>, width<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
                     strand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span><span class="op">)</span>, <span class="fl">200</span>, <span class="cn">TRUE</span><span class="op">)</span>,
                     feature_id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"ID%03d"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">200</span><span class="op">)</span><span class="op">)</span>
<span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html">DataFrame</a></span><span class="op">(</span>Treatment<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">c</a></span><span class="op">(</span><span class="st">"ChIP"</span>, <span class="st">"Input"</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>,
                     row.names<span class="op">=</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span>

<span class="fu"><a href="../reference/RangedSummarizedExperiment-class.html">SummarizedExperiment</a></span><span class="op">(</span>assays<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">counts</span><span class="op">)</span>,
                     rowRanges<span class="op">=</span><span class="va">rowRanges</span>, colData<span class="op">=</span><span class="va">colData</span><span class="op">)</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 200 6 
## metadata(0):
## assays(1): counts
## rownames: NULL
## rowData names(1): feature_id
## colnames(6): A B ... E F
## colData names(1): Treatment</code></pre>
<p>A <code>SummarizedExperiment</code> can be constructed with or without supplying a <code>DataFrame</code> for the <em>rowData</em> argument:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/RangedSummarizedExperiment-class.html">SummarizedExperiment</a></span><span class="op">(</span>assays<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">counts</span><span class="op">)</span>, colData<span class="op">=</span><span class="va">colData</span><span class="op">)</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 200 6 
## metadata(0):
## assays(1): counts
## rownames: NULL
## rowData names(0):
## colnames(6): A B ... E F
## colData names(1): Treatment</code></pre>
</div>
<div id="common-operations-on-summarizedexperiment" class="section level1">
<h1 class="hasAnchor">
<a href="#common-operations-on-summarizedexperiment" class="anchor"></a>Common operations on <code>SummarizedExperiment</code>
</h1>
<div id="subsetting" class="section level2">
<h2 class="hasAnchor">
<a href="#subsetting" class="anchor"></a>Subsetting</h2>
<ul>
<li>
<code>[</code> Performs two dimensional subsetting, just like subsetting a matrix or data frame.</li>
</ul>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># subset the first five transcripts and first three samples</span>
<span class="va">se</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 5 3 
## metadata(2): '' formula
## assays(1): counts
## rownames(5): ENSG00000000003 ENSG00000000005 ENSG00000000419
##   ENSG00000000457 ENSG00000000460
## rowData names(0):
## colnames(3): SRR1039508 SRR1039509 SRR1039512
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
<ul>
<li>
<code>$</code> operates on <code><a href="../reference/SummarizedExperiment-class.html">colData()</a></code> columns, for easy sample extraction.</li>
</ul>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span><span class="op">[</span>, <span class="va">se</span><span class="op">$</span><span class="va">cell</span> <span class="op">==</span> <span class="st">"N61311"</span><span class="op">]</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 64102 2 
## metadata(2): '' formula
## assays(1): counts
## rownames(64102): ENSG00000000003 ENSG00000000005 ... LRG_98 LRG_99
## rowData names(0):
## colnames(2): SRR1039508 SRR1039509
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
</div>
<div id="getters-and-setters" class="section level2">
<h2 class="hasAnchor">
<a href="#getters-and-setters" class="anchor"></a>Getters and setters</h2>
<ul>
<li>
<code><a href="../reference/RangedSummarizedExperiment-class.html">rowRanges()</a></code> / (<code><a href="../reference/SummarizedExperiment-class.html">rowData()</a></code>), <code><a href="../reference/SummarizedExperiment-class.html">colData()</a></code>, <code><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html">metadata()</a></code>
</li>
</ul>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span>, <span class="fl">5</span>, <span class="fl">3</span>, dimnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>, <span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>

<span class="va">dates</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RangedSummarizedExperiment-class.html">SummarizedExperiment</a></span><span class="op">(</span>assays<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">counts</span><span class="op">)</span>,
                              rowData<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html">DataFrame</a></span><span class="op">(</span>month<span class="op">=</span><span class="va">month.name</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>, day<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Subset all January assays</span>
<span class="va">dates</span><span class="op">[</span><span class="fu"><a href="../reference/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">dates</span><span class="op">)</span><span class="op">$</span><span class="va">month</span> <span class="op">==</span> <span class="st">"January"</span>, <span class="op">]</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 1 3 
## metadata(0):
## assays(1): counts
## rownames(1): A
## rowData names(2): month day
## colnames(3): A B C
## colData names(0):</code></pre>
<ul>
<li>
<code><a href="../reference/SummarizedExperiment-class.html">assay()</a></code> versus <code><a href="../reference/SummarizedExperiment-class.html">assays()</a></code> There are two accessor functions for extracting the assay data from a <code>SummarizedExperiment</code> object. <code><a href="../reference/SummarizedExperiment-class.html">assays()</a></code> operates on the entire list of assay data as a whole, while <code><a href="../reference/SummarizedExperiment-class.html">assay()</a></code> operates on only one assay at a time. <code>assay(x, i)</code> is simply a convenience function which is equivalent to <code>assays(x)[[i]]</code>.</li>
</ul>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code>## List of length 1
## names(1): counts</code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></code></pre></div>
<pre><code>##                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
## ENSG00000000003        679        448        873        408       1138
## ENSG00000000005          0          0          0          0          0
## ENSG00000000419        467        515        621        365        587
## ENSG00000000457        260        211        263        164        245
## ENSG00000000460         60         55         40         35         78</code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># assay defaults to the first assay if no i is given</span>
<span class="fu"><a href="../reference/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></code></pre></div>
<pre><code>##                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
## ENSG00000000003        679        448        873        408       1138
## ENSG00000000005          0          0          0          0          0
## ENSG00000000419        467        515        621        365        587
## ENSG00000000457        260        211        263        164        245
## ENSG00000000460         60         55         40         35         78</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">se</span>, <span class="fl">1</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></code></pre></div>
<pre><code>##                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
## ENSG00000000003        679        448        873        408       1138
## ENSG00000000005          0          0          0          0          0
## ENSG00000000419        467        515        621        365        587
## ENSG00000000457        260        211        263        164        245
## ENSG00000000460         60         55         40         35         78</code></pre>
</div>
<div id="range-based-operations" class="section level2">
<h2 class="hasAnchor">
<a href="#range-based-operations" class="anchor"></a>Range-based operations</h2>
<ul>
<li>
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">subsetByOverlaps()</a></code> <code>SummarizedExperiment</code> objects support all of the <code><a href="../reference/findOverlaps-methods.html">findOverlaps()</a></code> methods and associated functions. This includes <code><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">subsetByOverlaps()</a></code>, which makes it easy to subset a <code>SummarizedExperiment</code> object by an interval.</li>
</ul>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Subset for only rows which are in the interval 100,000 to 110,000 of</span>
<span class="co"># chromosome 1</span>
<span class="va">roi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="st">"1"</span>, ranges<span class="op">=</span><span class="fl">100000</span><span class="op">:</span><span class="fl">1100000</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">subsetByOverlaps</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">roi</span><span class="op">)</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 74 8 
## metadata(2): '' formula
## assays(1): counts
## rownames(74): ENSG00000131591 ENSG00000177757 ... ENSG00000272512
##   ENSG00000273443
## rowData names(0):
## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
</div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 RC (2021-05-10 r80294)
## Platform: aarch64-apple-darwin20 (64-bit)
## Running under: macOS Big Sur 11.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] SummarizedExperiment_1.22.0       SummarizedExperimentGHA_1.23.5003
##  [3] Biobase_2.52.0                    GenomicRanges_1.45.0001          
##  [5] GenomeInfoDb_1.28.4               IRanges_2.26.0                   
##  [7] S4Vectors_0.30.2                  BiocGenerics_0.38.0              
##  [9] MatrixGenerics_1.4.3              matrixStats_0.61.0               
## [11] BiocStyle_2.20.2                 
## 
## loaded via a namespace (and not attached):
##  [1] highr_0.9              bslib_0.3.1            compiler_4.1.0        
##  [4] BiocManager_1.30.16    jquerylib_0.1.4        XVector_0.32.0        
##  [7] zlibbioc_1.38.0        bitops_1.0-7           tools_4.1.0           
## [10] digest_0.6.28          lattice_0.20-45        jsonlite_1.7.2        
## [13] evaluate_0.14          memoise_2.0.0          rlang_0.4.12          
## [16] Matrix_1.3-4           DelayedArray_0.19.0    yaml_2.2.1            
## [19] pkgdown_1.6.1          xfun_0.27              fastmap_1.1.0         
## [22] GenomeInfoDbData_1.2.6 stringr_1.4.0          knitr_1.36            
## [25] desc_1.4.0             fs_1.5.0               sass_0.4.0            
## [28] systemfonts_1.0.3      grid_4.1.0             rprojroot_2.0.2       
## [31] R6_2.5.1               textshaping_0.3.6      rmarkdown_2.11        
## [34] bookdown_0.24          magrittr_2.0.1         htmltools_0.5.2       
## [37] ragg_1.1.3             stringi_1.7.5          RCurl_1.98-1.5        
## [40] cachem_1.0.6           crayon_1.4.1</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Bioconductor Maintainer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
