<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Make a RangedSummarizedExperiment object from an ExpressionSet and
       vice-versa — makeSummarizedExperimentFromExpressionSet • SummarizedExperimentGHA</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Make a RangedSummarizedExperiment object from an ExpressionSet and
       vice-versa — makeSummarizedExperimentFromExpressionSet" />
<meta property="og:description" content="Coercion between RangedSummarizedExperiment and
  ExpressionSet is supported in both directions.
For going from ExpressionSet to
  RangedSummarizedExperiment, the
  makeSummarizedExperimentFromExpressionSet function is also
  provided to let the user control how to map features to ranges." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SummarizedExperimentGHA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.23.5003</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Extensions.html">Extending the SummarizedExperiment class</a>
    </li>
    <li>
      <a href="../articles/SummarizedExperiment.html">_SummarizedExperiment_ for Coordinating Experimental Assays, Samples, and Regions of Interest</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/vjcitn/SummarizedExperimentGHA">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Make a RangedSummarizedExperiment object from an ExpressionSet and
       vice-versa</h1>
    
    <div class="hidden name"><code>makeSummarizedExperimentFromExpressionSet.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Coercion between <a href='RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment</a> and
  <a href='https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html'>ExpressionSet</a> is supported in both directions.</p>
<p>For going from <a href='https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html'>ExpressionSet</a> to
  <a href='RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment</a>, the
  <code>makeSummarizedExperimentFromExpressionSet</code> function is also
  provided to let the user control how to map features to ranges.</p>
    </div>

    <pre class="usage"><span class='fu'>makeSummarizedExperimentFromExpressionSet</span><span class='op'>(</span><span class='va'>from</span>,
                                          mapFun<span class='op'>=</span><span class='va'>naiveRangeMapper</span>,
                                          <span class='va'>...</span><span class='op'>)</span>

<span class='co'>## range mapping functions</span>
<span class='fu'>naiveRangeMapper</span><span class='op'>(</span><span class='va'>from</span><span class='op'>)</span>
<span class='fu'>probeRangeMapper</span><span class='op'>(</span><span class='va'>from</span><span class='op'>)</span>
<span class='fu'>geneRangeMapper</span><span class='op'>(</span><span class='va'>txDbPackage</span>, key <span class='op'>=</span> <span class='st'>"ENTREZID"</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>from</th>
      <td><p>An <a href='https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html'>ExpressionSet</a> object.</p></td>
    </tr>
    <tr>
      <th>mapFun</th>
      <td><p>A function which takes an <a href='https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html'>ExpressionSet</a> object and
    returns a GRanges, or GRangesList object which
    corresponds to the genomic ranges used in the ExpressionSet.  The
    <a href='https://rdrr.io/r/base/colnames.html'>rownames</a> of the returned <a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a>
    are used to match the <a href='https://rdrr.io/pkg/Biobase/man/featureNames.html'>featureNames</a> of the
    <a href='https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html'>ExpressionSet</a>.</p>
<p>The <code>naiveRangeMapper</code> function is used by default.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments passed to <code>mapFun</code>.</p></td>
    </tr>
    <tr>
      <th>txDbPackage</th>
      <td><p>A character string with the Transcript Database to use for the mapping.</p></td>
    </tr>
    <tr>
      <th>key</th>
      <td><p>A character string with the Gene key to use for the mapping.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>makeSummarizedExperimentFromExpressionSet</code> takes an
  <a href='https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html'>ExpressionSet</a> object as input and a <em>range mapping
  function</em> that maps the features to ranges.  It then returns a
  <a href='RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment</a> object that corresponds to the input.</p>
<p>The range mapping functions return a GRanges object, with the
  <code>rownames</code> corresponding to the <a href='https://rdrr.io/pkg/Biobase/man/featureNames.html'>featureNames</a> of
  the <a href='https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html'>ExpressionSet</a> object.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Jim Hester, <a href='james.f.hester@gmail.com'>james.f.hester@gmail.com</a></p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p></p><ul>
<li><p><a href='RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment</a> objects.</p></li>
<li><p><a href='https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html'>ExpressionSet</a> objects in the <span class="pkg">Biobase</span> package.</p></li>
<li><p><a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>TxDb</a> objects in the <span class="pkg">GenomicFeatures</span>
          package.</p></li>
</ul>
</div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='r-in'><span class='co'>## ---------------------------------------------------------------------</span></span>
<span class='r-in'><span class='co'>## GOING FROM ExpressionSet TO SummarizedExperiment</span></span>
<span class='r-in'><span class='co'>## ---------------------------------------------------------------------</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>sample.ExpressionSet</span>, package<span class='op'>=</span><span class='st'>"Biobase"</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># naive coercion</span></span>
<span class='r-in'><span class='fu'>makeSummarizedExperimentFromExpressionSet</span><span class='op'>(</span><span class='va'>sample.ExpressionSet</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: RangedSummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 500 26 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(3): experimentData annotation protocolData</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(2): exprs se.exprs</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames(500): AFFX-MurIL2_at AFFX-MurIL10_at ... 31738_at 31739_at</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(0):</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(26): A B ... Y Z</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(3): sex type score</span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/methods/as.html'>as</a></span><span class='op'>(</span><span class='va'>sample.ExpressionSet</span>, <span class='st'>"RangedSummarizedExperiment"</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: RangedSummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 500 26 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(3): experimentData annotation protocolData</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(2): exprs se.exprs</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames(500): AFFX-MurIL2_at AFFX-MurIL10_at ... 31738_at 31739_at</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(0):</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(26): A B ... Y Z</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(3): sex type score</span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/methods/as.html'>as</a></span><span class='op'>(</span><span class='va'>sample.ExpressionSet</span>, <span class='st'>"SummarizedExperiment"</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: SummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 500 26 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(3): experimentData annotation protocolData</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(2): exprs se.exprs</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames(500): AFFX-MurIL2_at AFFX-MurIL10_at ... 31738_at 31739_at</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(0):</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(26): A B ... Y Z</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(3): sex type score</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># using probe range mapper</span></span>
<span class='r-in'><span class='fu'>makeSummarizedExperimentFromExpressionSet</span><span class='op'>(</span><span class='va'>sample.ExpressionSet</span>, <span class='va'>probeRangeMapper</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>Accessing gene location information via 'CHR','CHRLOC','CHRLOCEND' is</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span>   deprecated. Please use a range based accessor like genes(), or select()</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span>   with columns values like TXCHROM and TXSTART on a TxDb or OrganismDb</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span>   object instead.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 'select()' returned 1:many mapping between keys and columns</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>148 probes could not be mapped.</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: RangedSummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 352 26 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(3): experimentData annotation protocolData</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(2): exprs se.exprs</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames(352): AFFX-HUMISGF3A/M97935_5_at AFFX-HUMISGF3A/M97935_MA_at</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   ... 31736_at 31737_at</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(0):</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(26): A B ... Y Z</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(3): sex type score</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># using the gene range mapper</span></span>
<span class='r-in'><span class='va'>se</span> <span class='op'>&lt;-</span> <span class='fu'>makeSummarizedExperimentFromExpressionSet</span><span class='op'>(</span></span>
<span class='r-in'>    <span class='va'>sample.ExpressionSet</span>,</span>
<span class='r-in'>    <span class='fu'>geneRangeMapper</span><span class='op'>(</span><span class='st'>"TxDb.Hsapiens.UCSC.hg19.knownGene"</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 'select()' returned 1:many mapping between keys and columns</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Registered S3 methods overwritten by 'SummarizedExperiment':</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>   method                                  from                   </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>   .DollarNames.SummarizedExperiment       SummarizedExperimentGHA</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>   .DollarNames.RangedSummarizedExperiment SummarizedExperimentGHA</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>   403 genes were dropped because they have exons located on both strands</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>   of the same reference sequence or on more than one reference sequence,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>   so cannot be represented by a single genomic range.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>   Use 'single.strand.genes.only=FALSE' to get all the genes in a</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>   GRangesList object, or use suppressMessages() to suppress this message.</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>171 probes could not be mapped.</span>
<span class='r-in'><span class='va'>se</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: RangedSummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 329 26 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(3): experimentData annotation protocolData</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(2): exprs se.exprs</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames(329): AFFX-HUMISGF3A/M97935_5_at AFFX-HUMISGF3A/M97935_5_at</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   ... 31736_at 31737_at</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(1): gene_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(26): A B ... Y Z</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(3): sex type score</span>
<span class='r-in'><span class='fu'><a href='SummarizedExperiment-class.html'>rowData</a></span><span class='op'>(</span><span class='va'>se</span><span class='op'>)</span>  <span class='co'># duplicate row names</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> DataFrame with 329 rows and 1 column</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                                gene_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                            &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> AFFX-HUMISGF3A/M97935_5_at        6772</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> AFFX-HUMISGF3A/M97935_5_at        6772</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> AFFX-HUMISGF3A/M97935_5_at        6772</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> AFFX-HUMISGF3A/M97935_5_at        6772</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> AFFX-HUMGAPDH/M33197_5_at         2597</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ...                                ...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 31733_at                          5024</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 31734_at                          3227</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 31735_at                         80316</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 31736_at                          6801</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 31737_at                            58</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## ---------------------------------------------------------------------</span></span>
<span class='r-in'><span class='co'>## GOING FROM SummarizedExperiment TO ExpressionSet</span></span>
<span class='r-in'><span class='co'>## ---------------------------------------------------------------------</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/utils/example.html'>example</a></span><span class='op'>(</span><span class='va'>RangedSummarizedExperiment</span><span class='op'>)</span>  <span class='co'># to create 'rse'</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; nrows &lt;- 200; ncols &lt;- 6</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; counts &lt;- matrix(runif(nrows * ncols, 1, 1e4), nrows)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rowRanges &lt;- GRanges(rep(c("chr1", "chr2"), c(50, 150)),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE+                      IRanges(floor(runif(200, 1e5, 1e6)), width=100),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE+                      strand=sample(c("+", "-"), 200, TRUE),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE+                      feature_id=sprintf("ID%03d", 1:200))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; colData &lt;- DataFrame(Treatment=rep(c("ChIP", "Input"), 3),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE+                      row.names=LETTERS[1:6])</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rse &lt;- SummarizedExperiment(assays=SimpleList(counts=counts),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE+                             rowRanges=rowRanges, colData=colData)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rse</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: RangedSummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 200 6 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(0):</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(1): counts</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames: NULL</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(1): feature_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(6): A B ... E F</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(1): Treatment</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; dim(rse)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] 200   6</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; dimnames(rse)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [[1]]</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> NULL</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [[2]]</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "A" "B" "C" "D" "E" "F"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; assayNames(rse)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "counts"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; head(assay(rse))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>             A        B          C        D        E        F</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1,] 4722.245 6566.958 9152.82308 1949.327 3179.950 2728.438</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [2,] 2826.128 4538.774 4992.65080 2735.593 8332.091 1554.358</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [3,] 8319.229 9138.503 2663.41726 9332.805 6347.146 8032.617</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4,] 5298.989 8171.587 9627.21522 3752.260 3154.935 3069.180</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [5,] 6627.937 8182.028   30.76941 4008.576 1366.718 2004.853</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [6,] 5236.918 4934.338 8488.34019 6210.713 1490.145 5358.842</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; assays(rse) &lt;- endoapply(assays(rse), asinh)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; head(assay(rse))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>             A        B        C        D        E        F</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1,] 9.153187 9.482953 9.814965 8.268387 8.757768 8.604632</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [2,] 8.639810 9.113559 9.208869 8.607251 9.721017 8.041965</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [3,] 9.719472 9.813399 8.580512 9.834438 9.448908 9.684413</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4,] 9.268419 9.701566 9.865496 8.923261 8.749870 8.722313</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [5,] 9.492196 9.702842 4.119932 8.989339 7.913315 8.296473</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [6,] 9.256636 9.197121 9.739596 9.427178 7.999776 9.279650</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rowRanges(rse)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 200 ranges and 1 metadata column:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>         seqnames        ranges strand |  feature_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>            &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>     [1]     chr1 913405-913504      - |       ID001</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>     [2]     chr1 397194-397293      - |       ID002</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>     [3]     chr1 466379-466478      - |       ID003</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>     [4]     chr1 936071-936170      + |       ID004</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>     [5]     chr1 736514-736613      + |       ID005</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>     ...      ...           ...    ... .         ...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [196]     chr2 279383-279482      + |       ID196</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [197]     chr2 765902-766001      - |       ID197</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [198]     chr2 129822-129921      - |       ID198</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [199]     chr2 921802-921901      - |       ID199</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [200]     chr2 866617-866716      - |       ID200</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rowData(rse)  # same as 'mcols(rowRanges(rse))'</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> DataFrame with 200 rows and 1 column</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>      feature_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>     &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1         ID001</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 2         ID002</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3         ID003</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 4         ID004</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 5         ID005</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ...         ...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 196       ID196</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 197       ID197</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 198       ID198</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 199       ID199</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 200       ID200</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; colData(rse)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> DataFrame with 6 rows and 1 column</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>     Treatment</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> A        ChIP</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> B       Input</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> C        ChIP</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> D       Input</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> E        ChIP</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> F       Input</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rse[ , rse$Treatment == "ChIP"]</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: RangedSummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 200 3 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(0):</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(1): counts</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames: NULL</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(1): feature_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(3): A C E</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(1): Treatment</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; ## cbind() combines objects with the same ranges but different samples:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rse1 &lt;- rse</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rse2 &lt;- rse1[ , 1:3]</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; colnames(rse2) &lt;- letters[1:ncol(rse2)] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; cmb1 &lt;- cbind(rse1, rse2)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; dim(cmb1)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] 200   9</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; dimnames(cmb1)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [[1]]</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> NULL</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [[2]]</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "A" "B" "C" "D" "E" "F" "a" "b" "c"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; ## rbind() combines objects with the same samples but different ranges:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rse1 &lt;- rse</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rse2 &lt;- rse1[1:50, ]</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rownames(rse2) &lt;- letters[1:nrow(rse2)] </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; cmb2 &lt;- rbind(rse1, rse2)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; dim(cmb2)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] 250   6</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; dimnames(cmb2)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [[1]]</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [19] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [37] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [55] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [73] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [91] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [109] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [127] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [145] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [163] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [181] ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  ""  "" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [199] ""  ""  "a" "b" "c" "d" "e" "f" "g" "h" "i" "j" "k" "l" "m" "n" "o" "p"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [217] "q" "r" "s" "t" "u" "v" "w" "x" "y" "z" NA  NA  NA  NA  NA  NA  NA  NA </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [235] NA  NA  NA  NA  NA  NA  NA  NA  NA  NA  NA  NA  NA  NA  NA  NA </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [[2]]</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "A" "B" "C" "D" "E" "F"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; ## Coercion to/from SummarizedExperiment:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; se0 &lt;- as(rse, "SummarizedExperiment")</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; se0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: SummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 200 6 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(0):</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(1): counts</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames: NULL</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(1): feature_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(6): A B ... E F</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(1): Treatment</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; as(se0, "RangedSummarizedExperiment")</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: RangedSummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 200 6 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(0):</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(1): counts</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames: NULL</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(1): feature_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(6): A B ... E F</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(1): Treatment</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; ## Setting rowRanges on a SummarizedExperiment object turns it into a</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; ## RangedSummarizedExperiment object:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; se &lt;- se0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; rowRanges(se) &lt;- rowRanges</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; se  # RangedSummarizedExperiment</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: RangedSummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 200 6 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(0):</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(1): counts</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames: NULL</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(1): feature_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(6): A B ... E F</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(1): Treatment</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; ## Sanity checks:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; stopifnot(identical(assays(se0), assays(rse)))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; stopifnot(identical(dim(se0), dim(rse)))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; stopifnot(identical(dimnames(se0), dimnames(rse)))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; stopifnot(identical(rowData(se0), rowData(rse)))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> RngdSE&gt; stopifnot(identical(colData(se0), colData(rse)))</span>
<span class='r-in'><span class='va'>rse</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> class: RangedSummarizedExperiment </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> dim: 200 6 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> metadata(0):</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assays(1): counts</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rownames: NULL</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> rowData names(1): feature_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colnames(6): A B ... E F</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> colData names(1): Treatment</span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/methods/as.html'>as</a></span><span class='op'>(</span><span class='va'>rse</span>, <span class='st'>"ExpressionSet"</span><span class='op'>)</span></span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>No assay named ‘exprs’ found, renaming counts to ‘exprs’.</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExpressionSet (storageMode: lockedEnvironment)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> assayData: 200 features, 6 samples </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   element names: exprs </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> protocolData: none</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> phenoData</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   sampleNames: A B ... F (6 total)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   varLabels: Treatment</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   varMetadata: labelDescription</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> featureData</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   featureNames: 1 2 ... 200 (200 total)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   fvarLabels: seqnames start ... feature_id (6 total)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   fvarMetadata: labelDescription</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> experimentData: use 'experimentData(object)'</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Annotation:  </span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Bioconductor Maintainer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


